File: 2025-1/sesion-de-clase-12 - 1/SoftProgDBManager/DBManager.cs
using System;
using System.Configuration;
using System.Data.Common;

namespace PUCP.Edu.Pe.SoftProg.Config {
    public class DBManager {
        private static DBManager instance;

        private string providerName;
        private string host;
        private int puerto;
        private string baseDatos;
        private string usuario;
        private string password;

        private DBManager() { 
        }

        public static DBManager GetInstance() {
            if (instance == null) {
                instance = new DBManager();
            }
            return instance;
        }

        public DbConnection GetConnection() {
            try {
                string cadenaConexion = this.CadenaConexion();
                DbProviderFactory factory = 
                    DbProviderFactories.GetFactory(this.providerName);

                var conexion = factory.CreateConnection();
                conexion.ConnectionString = cadenaConexion;
                conexion.Open();
                return conexion;
            }
            catch (Exception e) {
                Console.Error.WriteLine(e);
                throw;
            }
        }

        private string CadenaConexion() {
            this.providerName = ConfigurationManager.AppSettings["db.provider"];
            this.host = ConfigurationManager.AppSettings["db.host"];
            this.puerto = int.Parse(ConfigurationManager.AppSettings["db.puerto"]);
            this.baseDatos = ConfigurationManager.AppSettings["db.basedatos"];
            this.usuario = ConfigurationManager.AppSettings["db.usuario"];
            this.password = ConfigurationManager.AppSettings["db.password"];

            string cadenaConexion;
            switch (this.providerName) {
                case "System.Data.SqlClient":
                    cadenaConexion= $"Server={this.host},{this.puerto};Database={this.baseDatos};User Id={this.usuario};Password={this.password};TrustServerCertificate=True;";
                    break;
                case "MySql.Data.MySqlClient":
                    cadenaConexion = $"Server={this.host};Port={this.puerto};Database={this.baseDatos};Uid={this.usuario};Pwd={this.password};SslMode=none;";
                    break;
                default:
                    throw new NotSupportedException($"Proveedor no soportado: {this.providerName}");
            }

            return cadenaConexion;
        }
    }
}

File: 2026-2/sesion-de-clase-09/softprog/softprog-dbmanager/src/main/java/pe/edu/pucp/softprog/db/DBManager.java
package pe.edu.pucp.softprog.db;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.ResourceBundle;

public class DBManager {
    protected String servidor;
    protected int puerto;
    protected String esquema;
    protected String usuario;
    protected String password;

    private static DBManager instance;

    protected DBManager() {
    }

    protected DBManager(String servidor,
                        int puerto,
                        String esquema,
                        String usuario,
                        String password) {

        this.servidor = servidor;
        this.puerto = puerto;
        this.esquema = esquema;
        this.usuario = usuario;
        this.password = password;
    }

    public synchronized static DBManager getInstance() {
        if (instance == null) {
            ResourceBundle rb = ResourceBundle.getBundle("db");

            instance = new DBManager(
                rb.getString("servidor"),
                Integer.parseInt(rb.getString("puerto")),
                rb.getString("esquema"),
                rb.getString("usuario"),
                rb.getString("password")
            );
        }
        return instance;
    }

    public Connection getConnection() throws SQLException {
        return DriverManager.getConnection(
            "jdbc:mysql://" + servidor + ":" + puerto + "/" + esquema,
            usuario,
            password
        );
    }
}
